rust-toolchain.toml

```diff
@@ -0,0 +1,3 @@
+[toolchain]
+channel = "1.70.0"
+components = ["clippy", "rustfmt"]
```

Comment:
Can we remove these components from the toolchain file?

I previously asked the `rustup` team about this and got guidance that the toolchain file shouldn't be used for developer convenience. I'm also not sure how defining the components here interacts with `artichoke/setup-rust` action in CI.

See:

- https://github.com/rust-lang/rustup/issues/2686#issuecomment-1083831334
- https://github.com/rust-lang/rustup/issues/2686#issuecomment-1084179901